frontends/krita/krita_diff/script.py

```diff
@@ -426,24 +453,12 @@ def controlnet_transparency_mask_inserter(self, layers: list, mask_image):
         mask_selection = Selection()
         mask_selection.setPixelData(mask_ba, sx, sy, sw, sh)
 
-        for layer in layers: 
-            layer.setVisible(True)
-            self.doc.setActiveNode(layer)
-            self.doc.setSelection(mask_selection)
-            self.doc.waitForDone() # deals with race condition?
-            add_mask_action.trigger()
-
+        self.doc.setActiveNode(glayer)
```

Comment:
Hey. Are you still strugging with this? I noticed you can use `self.node` from the `Script` class to get the current node. So how about this workaround, assuming users will always have inpaint mask layer selected before creating the group layer (in your current set up):

```suggestion
        #self.doc.setActiveNode(glayer)
        self.doc.setSelection(mask_selection)
        self.doc.refreshProjection() # deals with race condition?
        add_mask_action.trigger()
        if self.node.type == "paintLayer" and self.node.childNodes().count() > 1:
            glayer.addChildNode(self.node.childNodes()[0])
        self.doc.waitForDone()
        self.doc.setSelection(orig_selection)
```

The previous code checks if current selected layer is an paintLayer and a child has been created (the transparency mask). Due to race conditions, transparency mask is created as child of current selected layer (inpaint mask layer). So we simply move it as a child of the group layer instead. This worked for me.

Reply:
This one didn't work, but I found an approach that did.  Since I'm doing the transparency mask at the start instead of after image generation, I can just set up a QTimer and wait until the group layer is done being created.  Gotta go over a few more case handling issues my testers ran into and will set as ready when that's done.